Assets/Plugins/iOS/Shopify/BuyTests/BuySerializableTests.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import XCTest
+import PassKit
+import ProductName
+
+class BuySerializableTests: XCTestCase {
+    
+    let emailAddress = "test_email@shopify.com"
+    let firstName    = "first name"
+    let lastName     = "last name"
+    let phone        = "123-456-7890"
+    let address1     = "80 Spadina Ave"
+    let address2     = "420 Wellington"
+    let city         = "Toronto"
+    let company      = "Shopify"
+    let country      = "Canada"
+    let province     = "ON"
+    let zip          = "A1B 2C3"
+    
+    // ----------------------------------
+    //  MARK: - PKPayment -
+    //
+    func testPKPaymentSerializable() {
+        
+        let billingContact        = createContact()
+        let shippingContact       = billingContact
+        let shippingMethod        = PKShippingMethod.init(label: "Free Shipping", amount: 0)
+        shippingMethod.identifier = "unique_id"
+        
+        let paymentMethod = MockPaymentMethod.init(displayName: "AMEX", network: .amex, type: .credit)
+        let token         = MockPaymentToken.init(paymentMethod: paymentMethod)
+        let payment       = MockPayment.init(token: token,
+                                             billingContact: billingContact,
+                                             shippingContact: shippingContact,
+                                             shippingMethod: shippingMethod)
+        
+        // Serialize and record data from result
+        let paymentDict         = payment.asDictionary()
+        let billingContactDict  = paymentDict[PKPaymentSerializedField.billingContact.rawValue]     as! [String: Any]
+        let shippingContactDict = paymentDict[PKPaymentSerializedField.shippingContact.rawValue]    as! [String: Any]
+        let tokenDict           = paymentDict[PKPaymentSerializedField.tokenData.rawValue]          as! [String: Any]
+        let shippingIdentifier  = paymentDict[PKPaymentSerializedField.shippingIdentifier.rawValue] as! String
+        
+        let tokenDataDict              = tokenDict[PKPaymentTokenSerializedField.paymentData.rawValue] as Any
+        let tokenData                  = try! JSONSerialization.data(withJSONObject: tokenDataDict)
+        let tokenTransactionIdentifier = tokenDict[PKPaymentTokenSerializedField.transactionIdentifier.rawValue] as! String
+        
+        assertEqualContact(billingContactDict,  contact: billingContact)
+        assertEqualContact(shippingContactDict, contact: shippingContact)
```

Comment:
Since `shippingContact` is also the `billingContact` in this case do we need to assert `shippingContactDict` and `shippingContact` are equal here since we're essentially doing the same thing in:
`assertEqualContact(billingContactDict,  contact: billingContact)`